packages/app/src/utils/signERC2612PermitPatch.ts

```diff
@@ -0,0 +1,23 @@
+import { signERC2612Permit } from "eth-permit";
+
+type ERC2612Permit = Awaited<ReturnType<typeof signERC2612Permit>>;
+
+const patchPermit = (permit: ERC2612Permit) => {
+  return {
+    ...permit,
+    v: permit.v === 0 ? 27 : permit.v === 1 ? 28 : permit.v,
+  };
+};
+
+export const signERC2612PermitPatched = async (
+  provider: any,
+  token: string,
+  owner: string,
+  spender: string,
+  value?: string | number,
+  deadline?: number | undefined,
+  nonce?: number | undefined
+): Promise<ERC2612Permit> => {
+  const result = await signERC2612Permit(provider, token, owner, spender, value, deadline, nonce);
+  return patchPermit(result);
```

Comment:
I'm generally not a fan of mutation. 

We could avoid a large diff here by having a helper that only takes in the current "v" value returned by signERC2612Permit and returns a fresh value which can then be used as a return value from the hook itself

Reply:
otherwise we always have to patch it in every call site (every hook), in my opinion this is a safer solution to prevent regressions.